plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/MachineProviderImpl.java

```diff
@@ -166,6 +171,11 @@ public MachineProviderImpl(DockerConnector docker,
         this.doForcePullOnBuild = doForcePullOnBuild;
         this.privilegedMode = privilegedMode;
         this.snapshotUseRegistry = snapshotUseRegistry;
+        if (Strings.isNullOrEmpty(cheUserId)) {
+            this.cheUserId = System.getenv("CHE_USER_ID") == null ? "" : System.getenv("CHE_USER_ID");
```

Comment:
To override value of this property env variable with special name should be used. Letters are capitalized, underscores are transformed into double underscores, dots into single underscores.
In that case injection of property doesn't need to use **System.getenv** at all. Developer injects property and Che bootstrapping sequence take care on reading env variable and injecting it as  property.